doc/design/multi_device_training.md

```diff
@@ -0,0 +1,47 @@
+# Multi-device training in PaddlePaddle
+
+## Why Multi-device
+On a typical device or system, there are multiple computing devices. In PaddlePaddle, the supported device types as of now, are CPU and GPU (we will be adding FPGA in the future too).
+
+If a PaddlePaddle operation has both CPU and GPU implementations, we decide which kernel to execute based on the device type.  
+Training deep learning models can be resource intensive. Even with a very powerful GPU, some models can take really long to train. This is obvious with deep learning models, especially recurrent models where the execution of each step depends on the execution and output of previous step.
```

Comment:
Training is a scene of using multi-device. In inference, we may also need to support multi-device. For example, FPGA is used for inference, but FPGA is not suitable for all operators. We may switch to CPU for some complex operators and then, switch back to FPGA.

Reply:
@QingshuChen and @shijiaxin Please kindly review this doc. We are planning to support multi-device feature for Paddle.

Reply:
Thanks for pointing this out. Will update the writeup.